libraries/AP_NavEKF/AP_NavEKF_Source.cpp

```diff
@@ -0,0 +1,431 @@
+/*
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_Math/AP_Math.h>
+#include <AP_DAL/AP_DAL.h>
+#include "AP_NavEKF_Source.h"
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_NavEKF_Source::var_info[] = {
+
+    // @Param: POSXY
+    // @DisplayName: Position Horizontal Source (Primary)
+    // @Description: Position Horizontal Source (Primary)
+    // @Values: 0:None, 3:GPS, 4:Beacon, 6:ExternalNav
+    // @User: Advanced
+    AP_GROUPINFO("POSXY", 1, AP_NavEKF_Source, _source[0].posxy, (int8_t)AP_NavEKF_Source::SourceXY::GPS),
+
+    // @Param: VELXY
```

Comment:
Maybe, the bonus of doing what you suggest is it would allow us to add more parameters to each "bank" (or whatever we call it) more easily in the future.  The downside is that I suspect we will already be reaching the limit of the depth of parameters.